### PR TITLE
Fix #31

### LINKED ISSUE #31: SPI only works with 4-8bit transfers
A transfer of a larger bit length (9 to 16bit) requires accessing 16bit words and while this can be configured, starting a 16 bit transfer doesn't set the `SPI_EVENT_INTERNAL_TRANSFER_COMPLETE` when required, probably due to still using 8bit lengths.

### PR DESCRIPTION
**Pull request by 0xc0170**
Need to get our new test which should confirm this fixes it, will append the result soon.

@salkinium @bogdanm

### CODE REVIEW

**Comment by salkinium** [comment]
+1 for now.

I will add an RFC on Error Handling and Reporting in HAL layers. There should not be an assert in these layers.

**Comment by 0xc0170** [comment]
@salkinium definitely , go for it

**Comment by bogdanm** [comment]
> I will add an RFC on Error Handling and Reporting in HAL layers. There should not be an assert in these layers.

That depends. What do you do, for example, if you ask for an UART speed that's not possible in hardware? I'd personally assert on this immediately, since there's very likely no way to recover from this error. For things that can be emulated in the software, I agree we shouldn't assert.

**Comment by 0xc0170** [comment]
@bogdanm save your energy for RFC :P review please the changeset

File: source/spi_api.c [hunk 1]
```diff
@@ -166,7 +167,14 @@ void spi_master_transfer(spi_t *obj, void *tx, size_t tx_length, void *rx, size_
 {
     (void)hint;
     MBED_ASSERT(hint == DMA_USAGE_NEVER); // only IRQ supported
+    MBED_ASSERT((obj->spi.bits > 8) && ((tx_length % 2) == 0) && ((rx_length % 2) == 0)); // alignment check for 16bit transfer
```
Comment (bogdanm) [inline source/spi_api.c]: Won't this fail for `obj->spi.bits == 8` ?

**Comment by 0xc0170** [comment]
@bogdanm corrected check for only 16 bit transfers

**Comment by bogdanm** [comment]
+1

**Pull request merged**